spacelift/resource_stack_dependency_reference.go

```diff
@@ -0,0 +1,179 @@
+	d.Set("input_name", query.StackDependencyReference.InputName)
+	return nil
+}
+
+func resourceStackDependencyReferenceRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var query struct {
+		Stack *struct {
+			Dependency *struct {
+				Reference *structs.StackDependencyReference `graphql:"reference(id: $reference_id)"`
+			} `graphql:"dependency(id: $dependency_id)"`
+		} `graphql:"stack(id: $stack_id)"`
+	}
+
+	idParts, diags := getStackDependencyReferenceIDParts(d)
+	if diags != nil {
+		return diags
+	}
+
+	variables := map[string]interface{}{
+		"stack_id":      toID(idParts[0]),
+		"dependency_id": toID(idParts[1]),
+		"reference_id":  toID(idParts[2]),
+	}
+
+	if err := meta.(*internal.Client).Query(ctx, "StackDependenciesReferenceRead", &query, variables); err != nil {
+		return diag.Errorf("could not query for stack dependency reference: %s", err)
+	}
+
+	d.Set("stack_dependency_id", path.Join(idParts[0], idParts[1]))
+	d.Set("output_name", query.Stack.Dependency.Reference.OutputName)
+	d.Set("input_name", query.Stack.Dependency.Reference.InputName)
+	return nil
+}
+
+func resourceStackDependencyReferenceUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var query struct {
+		StackDependencyReference structs.StackDependencyReference `graphql:"stackDependenciesUpdateReference(reference: $reference)"`
+	}
+
+	idParts, diags := getStackDependencyReferenceIDParts(d)
+	if diags != nil {
+		return diags
+	}
+
+	variables := map[string]interface{}{
+		"reference": structs.StackDependencyReferenceUpdateInput{
+			ID:         toID(idParts[2]),
+			OutputName: toString(d.Get("output_name")),
+			InputName:  toString(d.Get("input_name")),
+			Type:       toString("ENVIRONMENT_VARIABLE"),
+		},
+	}
+
+	if err := meta.(*internal.Client).Mutate(ctx, "StackDependenciesUpdateReference", &query, variables); err != nil {
+		return diag.Errorf("could not update stack dependency reference: %s", err)
+	}
+
+	d.Set("stack_dependency_id", path.Join(idParts[0], idParts[1]))
+	d.Set("output_name", query.StackDependencyReference.OutputName)
+	d.Set("input_name", query.StackDependencyReference.InputName)
+	return nil
+}
+
+func resourceStackDependencyReferenceDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var query struct {
+		StackDependencyReference *structs.StackDependencyReference `graphql:"stackDependenciesDeleteReference(id: $id)"`
+	}
+
+	idParts, diags := getStackDependencyReferenceIDParts(d)
+	if diags != nil {
+		return diags
+	}
+
+	variables := map[string]interface{}{
+		"id": toID(idParts[2]),
+	}
+
+	if err := meta.(*internal.Client).Mutate(ctx, "StackDependenciesDeleteReference", &query, variables); err != nil {
+		return diag.Errorf("could not delete stack dependency reference: %s", err)
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func getStackDependencyIDParts(d *schema.ResourceData) ([]string, diag.Diagnostics) {
```

Comment:
Really minor suggestion, but what about instead of returning a slice, return the components. That way it's really easy to understand the calling code because you'd be able to write something like this:

```go
stackID, stackDependencyID, diags := getStackDependencyIDParts(d)
```